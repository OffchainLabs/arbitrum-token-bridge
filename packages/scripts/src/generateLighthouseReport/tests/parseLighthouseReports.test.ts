import { describe, expect, it } from "vitest";

import report from "./__mocks__/output.json";
import {
  parseNavigationResults,
  parseSnapshotResults,
  parseTimespanResults,
} from "../parseLighthouseReports";
import { FlowResult } from "lighthouse";

const mock = report as unknown as FlowResult;

describe("ParseLighthouseReports", () => {
  it("Parse the navigation result", () => {
    const parsedReport = parseNavigationResults([mock.steps[0]]);
    expect(parsedReport).toMatchSnapshot();

    const mergedReports = parseNavigationResults([
      mock.steps[0],
      mock.steps[0],
      mock.steps[0],
    ]);
    expect(mergedReports).toEqual(parsedReport);
  });

  it("Parse the timesmapn result", () => {
    const parsedReport = parseTimespanResults([mock.steps[1]]);
    expect(parsedReport).toMatchSnapshot();

    const multipleReports = parseTimespanResults([
      mock.steps[1],
      mock.steps[1],
      mock.steps[1],
      mock.steps[1],
      mock.steps[1],
    ]);
    expect(multipleReports).toEqual(parsedReport);
  });

  it("Parse the snapshot result", () => {
    const parsedReport = parseSnapshotResults([mock.steps[2]]);
    expect(parsedReport).toMatchSnapshot();

    const multipleReports = parseSnapshotResults([
      mock.steps[2],
      mock.steps[2],
      mock.steps[2],
      mock.steps[2],
      mock.steps[2],
    ]);
    expect(multipleReports).toEqual(parsedReport);
  });
});
