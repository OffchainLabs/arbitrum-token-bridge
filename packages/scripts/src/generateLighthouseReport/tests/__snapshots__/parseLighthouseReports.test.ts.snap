// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ParseLighthouseReports > Parse the navigation result 1`] = `
{
  "accessibility": 90,
  "best_practices": 96,
  "cls": {
    "numericValue": 0,
    "score": 100,
  },
  "fcp": {
    "numericValue": 442.245,
    "score": 100,
  },
  "lcp": {
    "numericValue": 6386.566,
    "score": 3,
  },
  "performance": 36,
  "seo": 100,
  "speed": {
    "numericValue": 5786.938,
    "score": 1,
  },
  "tbt": {
    "numericValue": 2630,
    "score": 0,
  },
}
`;

exports[`ParseLighthouseReports > Parse the timesmapn result 1`] = `
{
  "best_practices": 95,
  "cls": {
    "numericValue": 0.104,
    "score": 89,
  },
  "inp": {
    "numericValue": 352,
    "score": 68,
  },
  "longTasks": {
    "durationMs": 2817.872,
    "total": 20,
  },
  "performance": 40,
  "tbt": {
    "numericValue": 1839.541,
    "score": 0,
  },
}
`;
